spring-batch-infrastructure/src/main/java/org/springframework/batch/item/data/MongoCursorItemReader.java

```diff
@@ -0,0 +1,297 @@
+
+    private CloseableIterator<? extends T> cursor;
+
+    public MongoCursorItemReader() {
+        super();
+        setName(ClassUtils.getShortName(MongoCursorItemReader.class));
+    }
+
+    /**
+     * Used to perform operations against the MongoDB instance. Also handles the mapping
+     * of documents to objects.
+     * @param template the MongoOperations instance to use
+     * @see MongoOperations
+     */
+    public void setTemplate(MongoOperations template) {
+        this.template = template;
+    }
+
+    /**
+     * The targetType of object to be returned for each {@link #read()} call.
+     * @param targetType the targetType of object to return
+     */
+    public void setTargetType(Class<? extends T> targetType) {
+        this.targetType = targetType;
+    }
+
+    /**
+     * @param collection Mongo collection to be queried.
+     */
+    public void setCollection(String collection) {
+        this.collection = collection;
+    }
+
+    /**
+     * A Mongo Query to be used.
+     * @param query Mongo Query to be used.
+     */
+    public void setQuery(Query query) {
+        this.query = query;
+    }
+
+    /**
+     * A JSON formatted MongoDB query. Parameterization of the provided query is allowed
+     * via ?&lt;index&gt; placeholders where the &lt;index&gt; indicates the index of the
+     * parameterValue to substitute.
+     * @param queryString JSON formatted Mongo query
+     */
+    public void setQuery(String queryString) {
+        this.queryString = queryString;
+    }
+
+    /**
+     * {@link List} of values to be substituted in for each of the parameters in the
+     * query.
+     * @param parameterValues values
+     */
+    public void setParameterValues(List<Object> parameterValues) {
+        Assert.notNull(parameterValues, "Parameter values must not be null");
+        this.parameterValues = parameterValues;
+    }
+
+    /**
+     * JSON defining the fields to be returned from the matching documents by MongoDB.
+     * @param fields JSON string that identifies the fields to sort by.
+     */
+    public void setFields(String fields) {
+        this.fields = fields;
+    }
+
+    /**
+     * {@link Map} of property
+     * names/{@link org.springframework.data.domain.Sort.Direction} values to sort the
+     * input by.
+     * @param sorts map of properties and direction to sort each.
+     */
+    public void setSort(Map<String, Sort.Direction> sorts) {
+        Assert.notNull(sorts, "Sorts must not be null");
+        this.sort = convertToSort(sorts);
+    }
+
+    /**
+     * JSON String telling MongoDB what index to use.
+     * @param hint string indicating what index to use.
+     */
+    public void setHint(String hint) {
+        this.hint = hint;
+    }
+
+    /**
+     * The size of batches to use when iterating over results.
+     * @param batchSize size the batch size to apply to the cursor
+     */
+    public void setBatchSize(Integer batchSize) {
+        this.batchSize = batchSize;
```

Comment:
Nitpicking: Should we take care of `batchSize` when it is 0 or negative?
As I checked the spec of jpa-mongo

* Use {@literal 0 (zero)} for no limit. A <strong>negative limit</strong> closes the cursor after returning a single
* batch indicating to the server that the client will not ask for a subsequent one.

ref: https://github.com/spring-projects/spring-data-mongodb/blob/main/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/query/Query.java#L539-L546

Reply:
@hmhuan I'm sorry for the delay in responding.
Users would expect that this code is implemented using spring-data-mongodb, given the explicit use of `MongoOperations` as an argument in `MongoCursorItemReader`. Therefore, if `setBatchSize` handles 0 or negative values, I think it could reduce predictability.

Reply:
The type of the parameter should be the same as expected by the `Query` API. The setter should add the javadoc tag `@see Query#cursorBatchSize(int)` so the user knows that the value will be set on that field, and therefore knows the (default) values to use. I will take care of this change when merging the PR.